src/persist-client/src/batch.rs

```diff
@@ -360,6 +360,9 @@ impl<T: Timestamp + Codec64> BatchParts<T> {
                     move || {
                         let mut buf = Vec::new();
                         batch.encode(&mut buf);
+
+                        // Drop batch as soon as we can to reclaim its memory.
+                        drop(batch);
```

Comment:
I'm guessing you noticed this just from eyeballing, not any observations in the benchmark?